### PR TITLE
Fix error when using importing ast with a empty `let` variable declaration

### LINKED ISSUE #14707: An issue arises when using the `--import-ast` option to process an AST containing inline assembly.
## Description

The `--import-ast` option of `solc` is failing to correctly recognize the AST generated by an empty `let` statement.

This bug prevents `solc` from passing the `test/libsolidity/semanticTests/inlineAssembly/inlineasm_empty_let.sol` test case.

## Environment

- Compiler version: commit `7df949edb22598e9d946660020c5390d0ce81bf8`
- Target EVM version (as per compiler settings): `shanghai`

## Steps to Reproduce

1. Generate the AST in JSON format: `solc --combined-json ast inlineasm_empty_let.sol > ast.json`
2. Compile using the `--import-ast` option: `solc --import-ast ast.json`

`solc` should report an error: `Error: Failed to import AST: Expected "nodeType" to be of type string.`

This appears to be due to the `createVariableDeclaration` function in the `libyul/AsmJsonImporter.cpp` file not handling an empty `value` appropriately.

### PR DESCRIPTION
**Pull request by matheusaaguiar**
Fix #14707.

### CODE REVIEW

File: test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
+{
+  "absolutePath": "a",
```
Comment (r0qs) [inline test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.json:2]: This looks a bit weird, shouldn't it be `test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.sol`?

Comment (matheusaaguiar) [reply test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.json:2]: I think I had asked about this before, but I don't really remember it now. But, anyways, the other tests don't include the full name, just `a` or `L` (for libraries).

Comment (r0qs) [reply test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.json:2]: Hum...right, I'm not aware of this, maybe @cameel or @nikola-matic can provide some more info. So the [test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration_parseOnly.json](https://github.com/ethereum/solidity/pull/14723/files#diff-2a812777bdf0b19c6cb7f80e1ef5b037d8836a35f61ff388ade623e22dd99779R2) should also have `a` as `absolutePath`, no?

Comment (matheusaaguiar) [reply test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.json:2]: Yes, I corrected that already in my local branch. The only problem now is different offsets in the source locations generated ... I haven't figured what `soltest` does that it's different from running `solc --ast-compact-json`.

Comment (matheusaaguiar) [reply test/libsolidity/ASTJSON/assembly/empty_let_variable_declaration.json:2]: I have fixed the tests using the output of `soltest`.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**